experimental/dds/tree2/src/test/shared-tree-core/editManager.spec.ts

```diff
@@ -472,18 +472,19 @@ describe("EditManager", () => {
 		for (const scenario of buildScenario([], meta)) {
 			// Uncomment the code below to log the titles of generated scenarios.
 			// This is helpful for creating a unit test out of a generated scenario that fails.
-			// const title = scenario
-			// 	.map((s) => {
-			// 		if (s.type === "Pull") {
-			// 			return `Pull(${s.seq}) from:${s.from} ref:${s.ref}`;
-			// 		} else if (s.type === "Ack") {
-			// 			return `Ack(${s.seq})`;
-			// 		}
-			// 		return `Push(${s.seq})`;
-			// 	})
-			// 	.join("|");
-			// console.debug(title);
-			runUnitTestScenario(undefined, scenario);
+			const title = scenario
+				.map((s) => {
+					if (s.type === "Pull") {
+						return `Pull(${s.seq}) from:${s.from} ref:${s.ref}`;
+					} else if (s.type === "Ack") {
+						return `Ack(${s.seq})`;
+					}
+					return `Push(${s.seq})`;
+				})
+				.join("|");
+			it(title, () => {
```

Comment:
While that makes it easier to run and debug individual tests, it makes the run-time prohibitively high.
If you really want to make it easier to toggle between the two, you can try to do something like the
```
const [outerFixture, innerFixture] = individualTests
				? [describe, it]
				: [it, (title: string, fn: () => void) => fn()];
```
In this PR: https://github.com/microsoft/FluidFramework/pull/15745

Reply:
I made this change because this test takes about 3 seconds to run and it's timing out locally. I guess I could also change the local timeout for the tests since Noah mentioned that it's 4 seconds instead of 2. @noencke thoughts?

Reply:
Actually, the reason this change is here is because the combinatorial test now exceeds the 2000 ms test timeout, so the goal here is to make test times _lower_ :D But I suppose you mean that the overhead of doing many `it()`s ends up being slow too?

I do think we should break this up in _some_ way; maybe we can break it up into larger batches to avoid hitting test timeouts. Even on a fast machine _before_ this PR, this combinatorial test is easily within the "danger zone" (e.g. one of my PRs changed it to run each subtest twice instead of once, and that exceeded the timeout on my desktop).

Jenn has confirmed that this PR is not making this test an order of magnitude longer or anything; it's just "a little bit slower", but that's enough to trigger the timeout.

Reply:
The overhead of many many call to `it()` is definitely an issue. I take that this PR makes the test timeout even when we don't call `it()`, is that right?

Reply:
If you use:
`yeild* buildScenario(scenario, meta);`
instead of:
```
for (const built of buildScenario(scenario, meta)) {
	yield built;
}
```
Is it any better?

Reply:
It still feels wrong though: either we don't mind having longer tests, or we do, in which case breaking it up is kind of cheating. I think an argument could be made that 2 seconds is just too low: if the goal is to prevent a stuck/buggy test from just holding CI back too long then, 10 seconds seems fine since it'll a rare occurence. I can also see how we don't really need to be testing this on every PR, so I'd be OK turning down the number of steps from 5 to 4 and making a work item to add a test that just runs as part of the stress pipeline.

Reply:
I view the 2-second limit to mean "a test should take _way_ less time than a second to run (i.e. they should be on the order of milliseconds at worst)". But some tests are categorically different "long tests" (e.g. fuzz tests, integration tests with real network/servers/UI, or stress tests) and those get to have increased timeouts but have to go in a special place (CI maybe, non-local). I think we could argue that this is a "fuzz test" of sorts (it's not fuzzy, but it's kind of the same idea), and so it deserves a one-off longer timeout and should be moved into the stress pipeline like you said.

Having it run locally has been nice, since it has caught bugs more than once for me and if it was only on CI then I would have at best not seen the failure for another 20 minutes and at worst done additional work in my PR that would only later be invalidated by observing this bug on CI and having to take a different approach. But maybe the solution to that is just that I should run it locally manually when I know I am changing things in EditManager (can we make it more discoverable for devs?).

So, sure, let's turn down the steps for now, that's a good idea. I don't think breaking into batches is necessarily cheating though. I view the ideal solution to be what Jenn did above (make 5000 tests), in which case the problem is not with our timeouts, but rather that `it` is too slow, or that the TestExplorerUI isn't virtualized, or whatever. The batching is meant to circumvent that problem, not the timeout problem. The way I see it, we really do have 5000 tests here and the fact that they were in a single `it` block was already not what we'd want. But again, if we look at this as a "fuzz test", then I think it's different and batches don't make sense here.

Reply:
Yeah, it is quite tricky to think about. I agree that `it()` being too slow is its own problem and that batching is a fine workaround for it. If we ignore that and consider those tests as individual tests, then we're left with the question of how many of those does it make sense to run in a given context. I'd advocate for running a few whenever someone just wants to run all the tests, more when gaiting a PR, and much more in stress tests and when testing a change in `EditManager`. Unfortunately our tooling only lets us do so much. I think I'd be happy with just 2 flavors: "run all" and CI do just a few, while the stress pipeline should do many and whatever the stress pipeline does should easily be runnable locally when testing changes to `EditManager`.

Reply:
I've lowered the steps and added this work item for this work: https://dev.azure.com/fluidframework/internal/_workitems/edit/4557